agent/lib_guzzle6.c

```diff
@@ -342,14 +342,62 @@ const zend_function_entry nr_guzzle6_requesthandler_functions[]
 
 /* }}} */
 
+static char* nr_guzzle_version(zval* app TSRMLS_DC) {
+  char* retval = NULL;
+  zval* version = NULL;
+  zend_class_entry* ce = NULL;
+
+  if (0 == nr_php_is_zval_valid_object(app)) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application object is invalid",
+                     __func__);
+    return NULL;
+  }
+
+  ce = Z_OBJCE_P(app);
+  if (NULL == ce) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application has NULL class entry",
+                     __func__);
+    return NULL;
+  }
+
+  if (NULL == nr_php_get_class_constant(ce, "VERSION")) {
+    // If full version does not exist, then we will send empty string
+    return " ";
+  } else {
+    version = nr_php_get_class_constant(ce, "VERSION");
+  }
+
+  if (NULL == version) {
+    nrl_verbosedebug(NRL_FRAMEWORK, "%s: Application does not have VERSION",
+                     __func__);
+    return NULL;
+  }
+
+  if (nr_php_is_zval_valid_string(version)) {
+    retval = nr_strndup(Z_STRVAL_P(version), Z_STRLEN_P(version));
+  } else {
+    nrl_verbosedebug(NRL_FRAMEWORK,
+                     "%s: expected VERSION be a valid string, got type %d",
+                     __func__, Z_TYPE_P(version));
+  }
```

Comment:
Should this just return " " instead of falling through and returning `NULL` at the end?

Reply:
I think it's fine to return `NULL` because the `nr_php_packages_create()` function checks if the version is `NULL` and if it is, then it will set the version to an empty string: https://github.com/newrelic/newrelic-php-agent/blob/65ca1de83400be2c7bc6bc0dbce28b1ff56ecd1d/axiom/nr_php_packages.c#L42